polynote-kernel/src/main/scala/polynote/config/PolynoteConfig.scala

```diff
@@ -37,7 +37,22 @@ final case class PolynoteConfig(
 
 object PolynoteConfig {
   implicit val encoder: ObjectEncoder[PolynoteConfig] = deriveEncoder
-  implicit val decoder: Decoder[PolynoteConfig] = deriveDecoder
+  implicit val decoder: Decoder[PolynoteConfig] = deriveDecoder[PolynoteConfig].prepare {
```

Comment:
So, this decoder now does side effects? 😬 I think a better approach might be to just always parse a pre-defined base if it's present. Another approach that some unix apps take is to read an entire directory of config files, like `config.d`, and apply them in alphabetical order.

Reply:
- [ ] move this logic out of the decoder.

Reply:
@jeremyrsmith completely reworked this.